DarkSky.agent.lib.nut

```diff
@@ -10,43 +10,49 @@ class DarkSky {
     // It is up to your application to extract the data you require
 
     // Written by Tony Smith (@smittytone)
-    // Copyright Electric Imp, Inc. 2016
+    // Copyright Electric Imp, Inc. 2016-18
     // License: MIT
 
+    static VERSION = "2.0.0";
     static FORECAST_URL = "https://api.darksky.net/forecast/";
-    static VERSION = [1,0,1];
 
     _apikey = null;
     _units = null;
     _lang = null;
     _debug = false;
 
     constructor (key = null, debug = false) {
-        if (imp.environment() != ENVIRONMENT_AGENT) {
-            server.error("DarkSky class must be instantiated by the agent");
-            return null;
-        }
-
-        if (key == "" || key = null) {
-            server.error("DarkSky class requires an API key");
-            return null;
-        }
-
+        // Object constructor
+        // PARAMETERS
+        //   1. DarkSky API Key as a string (required)
+        //   2. Debugging flag (optional; default: false)
+        // RETURNS
+        //   DarkSky instance, or throws on error
+
+        /// Check for instantiation parameter errors
+        if (imp.environment() != ENVIRONMENT_AGENT) throw "DarkSky class must be instantiated by the agent";
+        if (key == "" || key = null) throw "DarkSky class requires an API key";
+        if (typeof key != "string") throw "DarkSky class requires an API key supplied as a string";
+
+        // Set instance properties
+        if (typeof debug != "bool") debug = false;
         _debug = debug;
-        _apikey = key;
         _units = "auto";
+        _apikey = key;
     }
 
     function forecastRequest(longitude = 999, latitude = 999, callback = null) {
```

Comment:
... and why don't we change it here?

Reply:
These should be floats, so I'll update to be consistent.